### PR TITLE
Corr plot module

### PR DESCRIPTION
**Pull request by IKrukov-HORIS**

### CODE REVIEW

File: python-package/lets_plot/bistro/corr.py [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+            if is_numeric_values(values):
+                res += 1
+
+        return res
+
+    def _add_common_params(self, plot):
+        _COLUMN_WIDTH = 60
+        _MIN_PLOT_WIDTH = 400
+        _MAX_PLOT_WIDTH = 900
+        _PLOT_PROPORTION = 3.0 / 4.0
```
Comment (alshan) [inline python-package/lets_plot/bistro/corr.py:307]: This ratio is likely depends on presence of legend.

File: python-package/lets_plot/bistro/corr.py [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+
+        plot += theme(axis_title=element_blank(),
+                      legend_title=element_blank(),
+                      axis_line_x=element_blank(),
+                      axis_line_y=element_blank())
+
+        plot += coord_fixed()
+        plot += scale_size_identity(name="", na_value=0)
+        plot += self._color_scale
```
Comment (alshan) [inline python-package/lets_plot/bistro/corr.py:316]: 'scale_color..' is not enough, tiles' use 'scale_fill..' ?

File: python-package/lets_plot/bistro/corr.py [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+            return type
+
+        res = type if type else "full"
+
+        if self._reverse_y:
+            res = _reverse_type(res)
+
+        return res
+
+    def _set_diverging_palette(self, palette):
```
Comment (alshan) [inline python-package/lets_plot/bistro/corr.py:350]: Let's rename to _set_brewer_palette.

File: python-package/lets_plot/bistro/corr.py [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+        self._color_scale = scale_color_brewer(name='Correlation',
+                                               type='div',
```
Comment (alshan) [inline python-package/lets_plot/bistro/corr.py:352]: Remove `type='div'`

File: python-package/lets_plot/bistro/corr.py [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+        This method create PlotSpec and returns it.
+
+        Returns
+        -------
+            self
+        """
+
+        layers = []
+
+        if self._tiles_layer:
```
Comment (alshan) [inline python-package/lets_plot/bistro/corr.py:155]: Let's use API: 
```
plot = ggplot(self._data)
if self._tiles_layer:
  plot += self._tiles_layer

...
```

**Review by alshan — CHANGES_REQUESTED** [review]

File: python-package/lets_plot/bistro/corr.py (hunk 2 above)
Comment (IKrukov-HORIS) [reply python-package/lets_plot/bistro/corr.py:314]: Tiles are implemented as point_geom with shape=15. Do we need scale_fill in this case?

File: python-package/lets_plot/bistro/corr.py [hunk 6]
```diff
@@ -48,14 +48,13 @@ def __init__(self, data, show_legend=None, flip=True):
         self._show_legend = show_legend
         self._format = '.2f'
         self._reverse_y = flip if flip else False
-        self._text_color = None
-        self._tiles_layer = None
-        self._points_layer = None
-        self._labels_layer = None
         self._color_scale = scale_color_gradient2(name='Correlation',
```
Comment (alshan) [inline python-package/lets_plot/bistro/corr.py:51]: 'Correlation' takes too much space on the plot, let's call it 'Corr'.

**Review by alshan — CHANGES_REQUESTED** [review]

File: python-package/lets_plot/bistro/corr.py (hunk 6 above)
Comment (IKrukov-HORIS) [reply python-package/lets_plot/bistro/corr.py]: Done

**Pull request merged**